runtime/themes/tokyonight.toml

```diff
@@ -1,96 +1,131 @@
-"ui.statusline.insert" = { fg = "black", bg = "green" }
-"ui.statusline.select" = { fg = "black", bg = "magenta" }
-"ui.text" = { fg = "foreground" }
-"ui.text.focus" = { fg = "cyan" }
-"ui.virtual.ruler" = { bg = "foreground_gutter" }
-"ui.virtual.whitespace" = { fg = "foreground_gutter" }
-"ui.virtual.inlay-hint" = { fg = "comment" }
-"ui.window" = { fg = "black" }
+"diff.delta" = { fg = "change" }
+"diff.delta.moved" = { fg = "blue" }
+"diff.minus" = { fg = "delete" }
+"diff.plus" = { fg = "add" }
 
-"error" = { fg = "red" }
-"warning" = { fg = "yellow" }
-"info" = { fg = "blue" }
-"hint" = { fg = "teal" }
-"diagnostic.error" = { underline = { style = "curl", color = "red" } }
-"diagnostic.warning" = { underline = { style = "curl", color = "yellow" } }
-"diagnostic.info" = { underline = { style = "curl", color = "blue" } }
-"diagnostic.hint" = { underline = { style = "curl", color = "teal" } }
-"special" = { fg = "orange" }
+error = { fg = "error" }
+hint = { fg = "hint" }
+info = { fg = "info" }
+warning = { fg = "yellow" }
+"diagnostic.error" = { underline = { style = "curl" } }
+"diagnostic.warning" = { underline = { style = "curl" } }
+"diagnostic.info" = { underline = { style = "curl" } }
+"diagnostic.hint" = { underline = { style = "curl" } }
 
-"markup.heading" = { fg = "cyan", modifiers = ["bold"] }
-"markup.list" = { fg = "cyan" }
-"markup.bold" = { fg = "orange", modifiers = ["bold"] }
-"markup.italic" = { fg = "yellow", modifiers = ["italic"] }
-"markup.strikethrough" = { modifiers = ["crossed_out"] }
-"markup.link.url" = { fg = "green" }
-"markup.link.text" = { fg = "light-gray" }
-"markup.quote" = { fg = "yellow", modifiers = ["italic"] }
-"markup.raw" = { fg = "cyan" }
+"ui.background" = { bg = "bg", fg = "fg" }
+"ui.cursor" = { modifiers = ["reversed"] }
+"ui.cursor.match" = { fg = "orange", modifiers = ["bold"] }
+"ui.cursorline.primary" = { bg = "bg-menu" }
+"ui.help" = { bg = "bg-menu", fg = "fg" }
+"ui.linenr" = { fg = "fg-gutter" }
+"ui.linenr.selected" = { fg = "fg-linenr" }
+"ui.menu" = { bg = "bg-menu", fg = "fg" }
+"ui.menu.selected" = { bg = "fg-selected" }
+"ui.popup" = { bg = "bg-menu", fg = "border-highlight" }
+"ui.selection" = { bg = "bg-highlight" }
+"ui.selection.primary" = { bg = "bg-highlight" }
+"ui.statusline" = { bg = "bg-menu", fg = "fg-dark" }
+"ui.statusline.inactive" = { bg = "bg-menu", fg = "fg-gutter" }
+"ui.statusline.normal" = { bg = "blue", fg = "bg", modifiers = ["bold"] }
+"ui.statusline.insert" = { bg = "light-green", fg = "bg", modifiers = ["bold"] }
+"ui.statusline.select" = { bg = "magenta", fg = "bg", modifiers = ["bold"] }
+"ui.text" = { bg = "bg", fg = "fg" }
+"ui.text.focus" = { bg = "bg-visual" }
+"ui.text.inactive" = { fg = "comment", modifiers = ["italic"] }
+"ui.text.info" = { bg = "bg-menu", fg = "fg" }
+"ui.virtual.ruler" = { bg = "fg-gutter" }
+"ui.virtual.whitespace" = { fg = "fg-gutter" }
+"ui.virtual.inlay-hint" = {  bg = "bg-inlay", fg = "teal" }
```

Comment:
As someone that uses `tokyo-night-storm` and inlay-hints, i'd be relatively unhappy in having so strong colours for `inlay-hints` they should imo be as unnoticeable as possible. That's why i also made them have no bg as well as its fg-color to be the same as comments (which imo is the best way to handle it).

Any insight as to why make them so bold in colour?

Reply:
I just think of it the other way around. I like to make them stand out a bit from the base code as it is not part of it and is useful hinting, so I kept the upstream colors.
I also completely understand your point of view, and would be happy to leave them as they were before, which is `"ui.virtual.inlay-hint" = { fg = "comment" }` as you described.

Reply:
I wouldn't mind giving your idea a test-drive, there's always the ability to revert it anyway in the future :)
Maybe i would base the decision on whether it should be colorful / plain based on how it is handled in other themes?

Reply:
After looking at the other themes, the inlay hints do seem to follow the more discreet approach, though there are both.
I'd say it's just a matter of personal preference.
As you seem OK with the change, let's leave it like this for now. There's always the option to change it back afterwards as you said!